optuna/storages/redis.py

```diff
@@ -0,0 +1,586 @@
+import copy
+from datetime import datetime
+import pickle
+
+from optuna._experimental import experimental
+from optuna import distributions
+from optuna import exceptions
+from optuna.storages import base
+from optuna.storages.base import DEFAULT_STUDY_NAME_PREFIX
+from optuna import structs
+from optuna import type_checking
+
+try:
+    import redis
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    _available = False
+
+if type_checking.TYPE_CHECKING:
+    from typing import Any  # NOQA
+    from typing import Dict  # NOQA
+    from typing import List  # NOQA
+    from typing import Optional  # NOQA
+
+
+@experimental("1.3.0")
```

Comment:
Since we released [v1.3.0](https://github.com/optuna/optuna/releases/tag/v1.3.0) today, please bump up the version:
```suggestion
@experimental("1.4.0")
```